substrate-archive/src/archive.rs

```diff
@@ -137,6 +149,20 @@ impl<B, R, D, DB> Builder<B, R, D, DB> {
 		self.max_block_load = Some(max_block_load);
 		self
 	}
+
+	/// Set the folder and targets for tracing.
+	/// This tells substrate-archive to also store all state-traces resulting from the execution of blocks.
+	///
+	/// # Note
+	/// Traces will only be collected if a coexisting WASM binary
+	/// for the runtime version of the block being currently executed is available.
```

Comment:
Oh, I realize I must have missed an important piece here. I thought that we were collecting traces in native mode (and that this was the "easy" part), but that's not correct at all then? Hmm. Does this mean that as soon as we have a "hole" in the chain for which we do not have a WASM blob we don't get any state traces at all?

Reply:
Well native execution is a part of it, so even if the runtime that you're running does not have a WASM binary then tracing will work. But yeah there will be holes in traces if a wasm binary isn't present for X version. So far that means anything older than 8.25

Reply:
Printing out what runtimes we know of seems like a good QoL thing.
And a big, visible, log message when we can't collect anymore traces? Or even halt?

But why can't `archive` switch to the new runtime and chug along?

Reply:
The wasm binaries that have tracing enabled are kept in the repo, so whenever there's an upgrade I would have to compile that binary with the new version, push a commit, and then those running archive would need to upgrade to the newer version/pull the binary from the repo in order to have traces work again. By default substrate does not have wasm tracing-enabled binaries (the feature does not even officially exist in polkadot, it requires manual compilation).

Reply:
Ofc, I'm dumb, thank you for explaining (again).

We'll have to think about how we can make this process a bit less painful, perhaps we can ask @s3krit to build extra artifacts for us as part of the normal release pipeline and we can get "official" builds with wasm tracing enabled at some point. Then we can either script pulling them into this repo or even point directly at our releases.

Reply:
> Printing out what runtimes we know of seems like a good QoL thing.
> And a big, visible, log message when we can't collect anymore traces? Or even halt?

Can we capture ^^^ in issues so we don't stall this PR any longer?

Reply:
immortalized in #207